### PR TITLE
Batiallowed sites

### PR DESCRIPTION
**Pull request by batiFinkelshtein**
i add white list for the allwed sites

### CODE REVIEW

**Review by MalkySinowitz — APPROVED** [review]
looks great!

File: extension-ui/background.js [hunk 1]
```diff
@@ -1,84 +1,139 @@
...
-// Initialize cache when the extension is loaded
-chrome.runtime.onStartup.addListener(() => initializeBlockedSitesCache());
-chrome.runtime.onInstalled.addListener(() => initializeBlockedSitesCache());
+chrome.runtime.onStartup.addListener(() => initializeCaches());
+chrome.runtime.onInstalled.addListener(() => initializeCaches());
 
-function initializeBlockedSitesCache(callback) {
-  chrome.storage.local.get("blockedSites", (data) => {
+function initializeCaches(callback) {
+  chrome.storage.local.get(["blockedSites", "allowedSites", "isBlackList"], (data) => {
```
Comment (SariGlick) [inline extension-ui/background.js:9]: allowed sites aren't stored in local storage, they are saved in DB
please remove it
you can declare a list in variable for developing time, until we'll get the data from the server

File: extension-ui/background.js [hunk 2]
```diff
@@ -1,84 +1,139 @@
...
-          console.error("Error executing script: ", error);
-        });
-      });
+
+    if (isBlackList) {
+      if (blockedSitesCache.some(site => hostname.includes(site))) {
+        blockSite(details.tabId);
+      }
+    } else {
+      if (allowedSitesCache.some(site => hostname.includes(site))) {
```
Comment (SariGlick) [inline extension-ui/background.js:47]: check only  if (!allowedSitesCache.some(site => hostname.includes(site))) without the return;

File: extension-ui/background.js [hunk 3]
```diff
@@ -1,84 +1,139 @@
...
+
 chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
   if (request.action === 'addBlockedSite') {
     const hostname = request.hostname.toLowerCase();
-    ensureBlockedSitesCacheInitialized(() => {
+    ensureCachesInitialized(() => {
       if (!blockedSitesCache.includes(hostname)) {
         blockedSitesCache.push(hostname);
         chrome.storage.local.set({ blockedSites: blockedSitesCache }, () => {
+          console.log(`Added to blocked sites: ${hostname}`);
```
Comment (SariGlick) [inline extension-ui/background.js:82]: remove console logs

File: extension-ui/background.js [hunk 4]
```diff
@@ -1,84 +1,139 @@
...
+        });
+      } else {
+        sendResponse({ success: false, message: "Site is already blocked." });
+      }
+    });
+    return true;
+  }
+
+  if (request.action === 'addAllowedSite') {
+    const hostname = request.hostname.toLowerCase();
```
Comment (SariGlick) [inline extension-ui/background.js:93]: you can't add in extension allowed site, it comes only from profile in DB

File: extension-ui/background.js [hunk 5]
```diff
@@ -1,84 +1,139 @@
...
       }
     });
-    return true; // כדי להורות שהתגובה היא אסינכרונית
-  } else if (request.action === 'getBlockedSites') {
-    ensureBlockedSitesCacheInitialized(() => {
+    return true;
+  }
+
+  if (request.action === 'toggleMode') {
+    ensureCachesInitialized(() => {
```
Comment (SariGlick) [inline extension-ui/background.js:109]: you can't toggle the mode, it's up to the profile's status you got from DB

File: extension-ui/popup.js [hunk 6]
```diff
@@ -7,28 +6,60 @@ document.addEventListener('DOMContentLoaded', function () {
...
+        a.textContent = hostname;
+        a.target = "_blank";
+        li.appendChild(a);
         blockedSitesList.appendChild(li);
       });
     });
   });
 
+  // חדש - הצגת הרשימה הלבנה
+  allowedSitesBtn.addEventListener('click', function () {
```
Comment (SariGlick) [inline extension-ui/popup.js:40]: we dont have tab of white list

File: extension-ui/popup.scss [hunk 7]
```diff
@@ -1,15 +1,63 @@
...
+  height: 100vh;
+  background-color: #F0F0F0;
+}
+
+.container {
+  text-align: center;
+  background-color: #FFFFFF;
+  border-radius: 8px;
+  box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
+  padding: 20px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:15]: replace with %

File: extension-ui/popup.scss [hunk 8]
```diff
@@ -1,15 +1,63 @@
...
+  width: 300px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:16]: replace with %

File: extension-ui/popup.scss [hunk 9]
```diff
@@ -1,15 +1,63 @@
...
+}
+
+button {
+  background-color: #007BFF;
```
Comment (SariGlick) [inline extension-ui/popup.scss:20]: save all colors to variables

File: extension-ui/popup.scss [hunk 10]
```diff
@@ -1,15 +1,63 @@
...
+  color: #FFFFFF;
+  border: none;
+  border-radius: 4px;
+  padding: 10px 15px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:24]: replace with %

File: extension-ui/popup.scss [hunk 11]
```diff
@@ -1,15 +1,63 @@
...
+  margin: 5px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:25]: replace with %

File: extension-ui/popup.scss [hunk 12]
```diff
@@ -1,15 +1,63 @@
...
+  font-size: 16px;
+  transition: background-color 0.3s;
+}
+
+button:hover {
+  background-color: #0056b3;
+}
+
+form {
+  margin: 20px 0;
```
Comment (SariGlick) [inline extension-ui/popup.scss:36]: replace with %

File: extension-ui/popup.scss [hunk 13]
```diff
@@ -1,15 +1,63 @@
...
+}
+
+input[type="text"] {
+  padding: 8px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:40]: replace with %

File: extension-ui/popup.scss [hunk 14]
```diff
@@ -1,15 +1,63 @@
...
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  width: 80%;
+  margin: 5px 0;
```
Comment (SariGlick) [inline extension-ui/popup.scss:44]: replace with %

File: extension-ui/popup.scss [hunk 15]
```diff
@@ -1,15 +1,63 @@
...
+ul {
+  list-style-type: none;
+  padding: 0;
+}
+
+li {
+  background-color: #F8F9FA;
+  border: 1px solid #DEE2E6;
+  border-radius: 4px;
+  padding: 10px;
```
Comment (SariGlick) [inline extension-ui/popup.scss:61]: replace with %

File: extension-ui/popup.scss [hunk 16]
```diff
@@ -1,15 +1,63 @@
...
+  margin: 5px 0;
```
Comment (SariGlick) [inline extension-ui/popup.scss:62]: replace with %

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by MalkySinowitz — APPROVED** [review]
looks good

**Review by MalkySinowitz — APPROVED** [review]
great!

**Review by SariGlick — APPROVED** [review]

**Pull request merged**